packages/client-core/src/admin/components/locations/AddEditLocationModal.tsx

```diff
@@ -98,12 +98,17 @@ export default function AddEditLocationModal(props: {
   const screenSharingEnabled = useHookstate<boolean>(location?.locationSetting.screenSharingEnabled || true)
   const locationType = useHookstate(location?.locationSetting.locationType || 'public')
   const sceneThumbnailState = useHookstate(getMutableState(SceneThumbnailState))
+  const engineState = useHookstate(getMutableState(EngineState))
+  const sceneSettingsEntities = useHookstate([] as Entity[])
+
   // only use query if engine is running
-  let sceneSettingsEntities: Entity[] = []
-  if (getState(EngineState).viewerEntity !== UndefinedEntity) {
-    // no viewerEntity if not in Editor, running the engine
-    sceneSettingsEntities = useQuery([SceneSettingsComponent])
-  }
+  useEffect(() => {
+    if (engineState.viewerEntity.value === UndefinedEntity) return
+    sceneSettingsEntities.set(useQuery([SceneSettingsComponent]))
```

Comment:
not quite what i meant. it is as simple as

```ts
const sceneSettingsEntities = useQuery([SceneSettingsComponent])
const sceneSettingsEntity = engineState.viewerEntity.value ? sceneSettingsEntities[0] : UndefinedEntity
```

And I noticed it also looks like `SceneThumbnailState.uploadThumbnail` only takes one entity, not an array of entities, hence the change above. Please add an argument type to `SceneThumbnailState.uploadThumbnail`

Reply:
const sceneSettingsEntities = useQuery([SceneSettingsComponent])
this line causes issues in the multitenancy page, we just cant use the query at all , as we dont have an engine running the background , when on the MT dashboard

Reply:
in that case, we should not have this logic here at all. the multitenancy page must not even import any of the engine packages to ensure it loads fast. we need to figure out a different solution

Reply:
this needs to be fixed. useQuery in this context will throw an error